src/client/Microsoft.Identity.Client/Platforms/iOS/IntuneEnrollmentIdHelper.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Json;
+#if iOS
+using Foundation;
+#endif
+
+namespace Microsoft.Identity.Client.Platforms.iOS
+{
+    internal class IntuneEnrollmentIdHelper
+    {
+        const string EnrollmentIdKey = "intune_app_protection_enrollment_id_V1";
+
+        internal static string GetEnrollmentId(ICoreLogger logger)
+        {
+#if iOS
+            var keychainData = NSUserDefaults.StandardUserDefaults.StringForKey(EnrollmentIdKey);
+            if(!string.IsNullOrEmpty(keychainData))
+            {
+                try
+                {
+                    var enrollmentIDs = JsonConvert.DeserializeObject <EnrollmentIDs>(keychainData);
+
+                    return enrollmentIDs.EnrollmentIds[0].EnrollmentId;
+
+                }
+                catch (Exception ex)
+                {
+                    logger.Error($"Failed to retreive enrollmentID {ex.Message}");
```

Comment:
`logger.Error($"Failed to retreive enrollmentID {ex}");`

- this way .NET will print out not just the exception, but also the exception trace and inner exceptions.


Also, please use `logger.Error()` which takes an Exception parameter and be mindful of PII (exception messages can have PII)